File: android_project/folder_type/MyDiceApp/app/src/main/java/cox/tenisme/mydiceapp/MainActivity.java
package cox.tenisme.mydiceapp;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.daimajia.androidanimations.library.Techniques;
import com.daimajia.androidanimations.library.YoYo;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    Button btnRoll;
    ImageView diceImg1;
    ImageView diceImg2;
    MediaPlayer mp;

    int[] diceImages = {R.drawable.dice1, R.drawable.dice2, R.drawable.dice3, R.drawable.dice4,
            R.drawable.dice5, R.drawable.dice6};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnRoll = findViewById(R.id.btnRoll);
        diceImg1 = findViewById(R.id.diceImg1);
        diceImg2 = findViewById(R.id.diceImg2);
        mp = MediaPlayer.create(this, R.raw.dice_sound);

        btnRoll.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.i("MyDiceApp","주사위 버튼 눌렀음!");

                Random rand = new Random();
                int diceNumber = rand.nextInt(6);
                Log.i("MyDiceApp", ""+diceNumber);

                diceImg1.setImageResource(diceImages[diceNumber]);

                // 랜덤 숫자 따로 한 번 더 돌리고 그 숫자는 diceImg2에 활용하기
                diceNumber = rand.nextInt(6);
                Log.i("MyDiceApp", ""+diceNumber);

                diceImg2.setImageResource(diceImages[diceNumber]);

                mp.start();

                YoYo.with(Techniques.RollIn)
                        .duration(300) // 초(밀리세컨. 2000 = 2초)
                        .repeat(1)
                        .playOn(diceImg1);
                YoYo.with(Techniques.RollIn)
                        .duration(300)
                        .repeat(1)
                        .playOn(diceImg2);

            }
        });
    }
}

File: android_project/folder_type/LockScreen/app/src/main/java/com/tenisme/lockscreen/MainActivity.kt
package com.tenisme.lockscreen

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private lateinit var onBtn: Button
    private lateinit var offBtn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        onBtn = findViewById(R.id.onBtn)
        offBtn = findViewById(R.id.offBtn)

        onBtn.setOnClickListener {
            val intent = Intent(applicationContext, ScreenService::class.java)
            startService(intent)
            Log.d("LockScreen", "Start LockScreen")
        }
        offBtn.setOnClickListener {
            val intent = Intent(applicationContext, ScreenService::class.java)
            stopService(intent)
            Log.d("LockScreen", "Stop LockScreen")
        }
    }
}